pil-analyzer/src/statement_processor.rs

```diff
@@ -647,4 +653,29 @@ where
             functions,
         }
     }
+
+    pub fn process_trait_implementation(
+        &self,
+        trait_impl: parsed::TraitImplementation<parsed::Expression>,
+    ) -> TraitImplementation<Expression> {
+        let type_vars = trait_impl.type_scheme.vars.vars().collect();
+        let functions = trait_impl
+            .functions
+            .iter()
```

Comment:
```suggestion
            .into_iter()
```
(then you don't need to clone the name I think)